routing/cross_mwm_road_graph.cpp

```diff
@@ -187,16 +186,18 @@ void CrossMwmGraph::GetOutgoingEdgesList(BorderCross const & v,
 
   // Find income node.
   IngoingCrossNode ingoingNode;
-  CHECK(currentContext.FindIngoingNodeByPoint(v.toNode.point, ingoingNode), ());
-  if (ingoingNode.m_nodeId != v.toNode.node)
-  {
-    LOG(LDEBUG, ("Several nodes stores in one border point.", v.toNode.point));
-    currentContext.ForEachIngoingNode([&ingoingNode, &v](IngoingCrossNode const & node)
-                                      {
-                                        if (node.m_nodeId == v.toNode.node)
-                                          ingoingNode = node;
-                                      });
-  }
+  bool found = false;
+  auto findingFn = [&ingoingNode, &v, &found](IngoingCrossNode const & node)
+                                             {
+                                               if (node.m_nodeId == v.toNode.node)
+                                               {
+                                                 found = true;
+                                                 ingoingNode = node;
+                                               }
+                                             };
+  CHECK(currentContext.ForEachIngoingNodeNearPoint(v.toNode.point, findingFn), ());
```

Comment:
For some isolated islands there may be no ingoung/outgoing points, right? And CHECK() would crash the APP.